KosherCocoa/Library/Core/Calendar/Zmanim/KCComplexZmanimCalendar.m

```diff
@@ -584,12 +586,17 @@ - (nullable NSDate *) sofZmanBiurChametzMGA16Point1Degrees{
 - (nullable NSDate *)solarMidnight {
     KCComplexZmanimCalendar *clonedCalendar = [self copy];
     [clonedCalendar setWorkingDate:[self dateByAddingDays:1 toDate:clonedCalendar.workingDate]];
-    NSDate *sunset = [self sunset];
-    NSDate *sunrise = [self sunrise];
-    
-    return [[self sunset] dateByAddingTimeInterval:[self temporalHourFromSunrise:sunrise toSunset:sunset] *6];
+        
+    return [[self chatzos] dateByAddingTimeInterval:clonedCalendar.chatzos.timeIntervalSince1970 - [self chatzos].timeIntervalSince1970 / 2];
```

Comment:
KosherJava just updated his code to this way

Reply:
It covers larger range of areas on the globe

Reply:
> larger range of areas on the globe

As in the arctic circle?

Reply:
Yes